lab7/src/main/java/agh/ics/oop/AbstractComparator.java

```diff
@@ -0,0 +1,13 @@
+package agh.ics.oop;
+
+import java.util.Comparator;
+
+abstract class AbstractComparator implements Comparator<ElementOfBoundarySet> {
+    protected int calcCompareForX(Vector2d o1, Vector2d o2) {
```

Comment:
tu rozumiem zamysł żeby mieć obie te metody w klasach rozszerzających, ale w tej sytuacji jest to właściwie zestaw narzędzi, z których korzystają obie klasy, a nie nadrzędna logika nad nimi --> w takich sytuacjach lepiej zamiast dziedziczenia pójść w kompozycję i np. stworzyć sobie klasę `ComparatorUtil` , a tam dwie metody statyczne `calcCompareForX()` i `calcCompareForY()`. Unikając dziedziczenia sprawiamy, że kod jest bardziej elastyczny, bo dziedziczenie zawsze wiąże nas z jakąś jedną klasą bazową.